src/Nethermind/Nethermind.TxPool/SimpleRetryCache.cs

```diff
@@ -0,0 +1,135 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core.Caching;
+using Nethermind.Logging;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Nethermind.TxPool;
+
+/// <summary>
+/// Allows to announce request for a resource and track other nodes to request it from in case of timeout. The code does not request the resource initially, but does so on timeout.
+/// </summary>
+/// <typeparam name="TResourceId">Resource identifier</typeparam>
+/// <typeparam name="TNodeId">Node that can be queried for the resource</typeparam>
+public interface ISimpleRetryCache<TResourceId, TNodeId>
+    where TResourceId : struct, IEquatable<TResourceId>
+    where TNodeId : notnull, IEquatable<TNodeId>
+{
+    AnnounceResult Announced(TResourceId resourceId, TNodeId nodeId, Action request);
+    void Received(TResourceId resourceId);
+}
+public abstract class SimpleRetryCache
+{
+    public const int TimeoutMs = 2500;
+    public const int CheckMs = 300;
+}
+
+public class SimpleRetryCache<TResourceId, TNodeId> : SimpleRetryCache, ISimpleRetryCache<TResourceId, TNodeId>
+    where TResourceId : struct, IEquatable<TResourceId>
+    where TNodeId : notnull, IEquatable<TNodeId>
+{
+    private readonly ConcurrentDictionary<TResourceId, Dictionary<TNodeId, Action>> _retryRequests = new();
+    private readonly ConcurrentQueue<(TResourceId ResourceId, DateTimeOffset Expires)> expiringQueue = new();
```

Comment:
Maybe instead of tracking `_retryRequests` internally the `RetryCache` could just sound an event that retry is needed and it could be handled by `PooledTxsRequestor` which would actually retry? Would make it less complicated and removed the need for passing `Action`s. Made the separation of concerns better.